File: Bishop.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace chess
{
    class Bishop : Piece
    {
        public Bishop(Square position, string color)
        {
            Position = position;
            Color = color;
            Content = Color == "White" ? "\u2657" : "\u265D";
        }
        public Square Position { get; set; }
        public string Content { get; set; }
        public string Color { get; set; }
        public void Capture(Piece target)
        {
            throw new NotImplementedException();
        }

        public bool IsAvailableMove(Square target)
        {
            throw new NotImplementedException();
        }

        public void Move(Square target)
        {
            throw new NotImplementedException();
        }

        public void Remove()
        {
            throw new NotImplementedException();
        }
    }
}

File: chess-cross-platform/chess/bishop.cpp
#include <string>
#include <cmath>
#include "bishop.h"
#include "square.h"

Bishop::Bishop(Square* position, std::string color) {
  this->Position = position;
  this->Color = color;
  this->Content = Color == "White" ? "\u2657" : "\u265D";
}

Bishop::Bishop(){}
Bishop::~Bishop(){}

bool Bishop::IsAvailableMove(Square* target) {
  if (abs(target->Rank - Position->Rank) == abs(target->File - Position->File))
    {
      return true;
    }
  else return false;
}

bool Bishop::CanCapture(Square* target) {
  if (IsAvailableMove(target)) return true;
  else return false;
}

void Bishop::Move(Square* target) {
  this->Position->Rank = target->Rank;
  this->Position->File = target->File;
}

